### ISSUE TITLE
Hi Nathan!

### ISSUE THREAD

**Issue by arthurvanl** [nathan-vdp/stage-opdracht#1]
Ik zie dat je een repo hebt gemaakt m.b.t. de stage-opdracht. Heb je hulp nodig bij de back-end?

**Comment by nathan-vdp** [comment]
Ja, ik had gister contact met Eva over de opdracht. Zelf ben ik voornamelijk bezig met front-end en heb vrij weinig kennis van back-end en heb er nog niet echt mee gewerkt. Eva had erover dat jij misschien de back-end kan realiseren.

**Comment by arthurvanl** [comment]
Yes zal ik doen voor je! 

Ik stuur gwn wat javascript code is dat voldoende? Wel benieuwd waarin je het gaat realiseren

**Comment by nathan-vdp** [comment]
Ik denk dat javascript voldoende is.

**Comment by arthurvanl** [comment]
Oke is goed. Maak je het met svelte kit of react?

**Comment by nathan-vdp** [comment]
Svelte kit. Daar ben k het meest bekend mee

**Comment by arthurvanl** [comment]
Oke, prima om het in daarin te maken! Sta je er wel voor open om op je stage met react/nextjs aan de slag te gaan? Uiteraard helpen we hier gewoon bij

**Comment by nathan-vdp** [comment]
Ja zeker, sta altjid open om wat nieuws te leren.

**Comment by nathan-vdp** [comment]
Heb op school wel al een beetje gewerkt met React maar nog niet zodanig dat ik een project kan voltooien

**Comment by arthurvanl** [comment]
Oke top!, ik zal ergens in de middag het maken voor je!

**Comment by arthurvanl** [comment]
```js
const API_KEY = '7aaf5b2bc34d37ffa73bbb6a6ac6ead12400f8c374dc4efd41ccbc7aed460bbc',
API_URL = 'https://dev.api.mintycloud.nl/api/v2.1',
domain = 'arthurvanleeuwen',
extensions = ['nl', 'be', 'com', 'co.uk', 'uk', 'fr', 'de', 'net', 'org', 'xyz'],

search = extensions.map((x) => ({name: domain, extension: x})),
headers = new Headers();
headers.append('Authorization', `Basic ${API_KEY}`);
const response = await fetch(`${API_URL}/domains/search?with_price=true`, {method: 'POST', headers, body: JSON.stringify(search)});
if(!response.ok) {
    //! bekijk de request body of er misschien fouten zit
    console.log(search)
    console.log(response)
    throw new Error('Er ging iets verkeerd!')
} else {
    const data = await response.json();
    console.log(data);
}```

Heb ff snel iets gemaakt. API_KEY kan je alleen voor deze endpoint gebruiken (niet voor andere)

**Comment by nathan-vdp** [comment]
Helemaal top! Ik ga er vandaag en volgende week op focussen ik hou jouw gewoon op de hoogte en je ziet ook wat ik doe via de commits.

**Comment by nathan-vdp** [comment]
Hey Arthur, Ik ben best goed op weg, hoe zit het eigenlijk met de opdrachten als je niet alles afkrijgt? Ik ga vandaag proberen om de domeinen in een winkelwagen te zetten.

**Comment by arthurvanl** [comment]
Hi Nathan,

Ik zou gewoon alles proberen te maken. Wetende dat er andere potentiële stagiaires deze opdracht ook maken zal het dan waarschijnlijk dan lastiger worden. Maak gewoon de opdracht ze goed mogelijk. Mocht iets niet afgemaakt te zijn, onderbouw het gewoon goed zodat wij kunnen begrijpen waarom je het niet gelukt is.

**Comment by nathan-vdp** [comment]
Ik kom hier misschien een beetje te laat mee, maar ik heb dus niet in de repo gewerkt van jullie, maar in een eigen en een svelte project aangemaakt, was dat de bedoeling of moest ik die code van jouw in de repo zetten van jullie en verder werken met wat jullie al hadden gemaakt? Ik heb nu zegmaar een hele eigen website gemaakt.

**Comment by nathan-vdp** [comment]
![Screenshot (117)](https://github.com/nathan-vdp/stage-opdracht/assets/115833333/1319ea88-0c3a-4efd-b1b0-242a0233cf0b)
Dit zegmaar wat ik heb gemaakt

**Comment by arthurvanl** [comment]
> Ik kom hier misschien een beetje te laat mee, maar ik heb dus niet in de repo gewerkt van jullie, maar in een eigen en een svelte project aangemaakt, was dat de bedoeling of moest ik die code van jouw in de repo zetten van jullie en verder werken met wat jullie al hadden gemaakt? Ik heb nu zegmaar een hele eigen website gemaakt.

Is inderdaad de bedoeling dat je je eigen repo maakt. Volgens mij kan je zelfs geen code toevoegen aan onze repository. Zowel, moeten we daar even een aanpassing voor maken. Als je de opdracht af hebt gemaakt, zal ik er naar kijken.

**Comment by nathan-vdp** [comment]
Moet ik de opdracht vandaag inleveren?

**Comment by arthurvanl** [comment]
Ik snap nu waarom het fout ging. Ik had niet een extensie moeten toevoegen. Dit is wel wat een normale gebruiker zou proberen tho.

**Comment by nathan-vdp** [comment]
ja klopt, Ik zie idd wanneer je .nl toevoegt dat ie een error geeft, en voor de winkelwagen daar ben ik nu mee bezig. Ik doe het via de local storage als je klikt op "voeg toe" dan kreeg nu nog niks te zien maar als je naar "winkewagen" gaat staat je domein erbij aan je producten overzicht

**Comment by arthurvanl** [comment]
Ik was ook naar winkelwagen gegaan nadat ik op de "Voeg toe" knop drukte, maar die pagina was ook leeg...

**Comment by nathan-vdp** [comment]
Misschien git pull. Ik heb zojuist gepusht

**Comment by arthurvanl** [comment]
Het ziet er naar uit dat er inderdaad de localeStorage geupdate wordt. Maar er gebeurd vervolgens nog niks op de winkelwagen pagina. En de orders werkt ook niet (dan krijg ik ook nog een order in de console te zien (niet browser console))

**Comment by nathan-vdp** [comment]
Dit is hoe het bij mij eruit ziet, alleen ik ben er niet aan toe gekomen om een mock transactie te maken
![Screenshot (118)](https://github.com/nathan-vdp/stage-opdracht/assets/115833333/fcdf4c5e-098d-4ba0-830f-bdbadce61e90)
![Screenshot (119)](https://github.com/nathan-vdp/stage-opdracht/assets/115833333/760a5839-276b-4ded-a363-e675ecb54802)

**Comment by nathan-vdp** [comment]
ik heb ook gemerkt wanneer ik een willekeurige naam in type dat het dan een status active teruggeeft

**Comment by arthurvanl** [comment]
Ga nog maar even uitvogelen anders :). Ik ga nu ervandoor.

**Comment by nathan-vdp** [comment]
Nou ik wil hem bij deze inleveren, eigenlijk wouw ik hem vrijdag inleveren maar er zaten nog wat fouten in. 

Ik heb wel gemerkt dat ik veel te leren heb vooral wanneer het op API's aankomt. Dat wordt leuk als ik dat bij jullie kan leren ;). 

als er nog fouten in zitten hoor ik dat graag dan probeer ik ze te fixen.

**Comment by arthurvanl** [comment]
Het ziet er naar uit dat de volgende dingen niet werken (getest op firefox):

- Kan nog steeds niet een domein toevoegen aan de winkelmand
- Kan niet zoeken met een extensie (e.g. example.com)
- Als je naar de winkelmand bent geswitched, kan je niet terug gaan naar de main page
   - Dit kan wel wanneer je eerst de pagina refreshed en op de winkelmand pagina bent.

**Comment by nathan-vdp** [comment]
Nou dan heb ik nog wat te doen en,  betreft de sollicitatie process ik neem aan dat deze opdracht een vereist is dat je alles hebt gemaakt voor een potentiële sollicitatiegesprek?

**Comment by arthurvanl** [comment]
Ja, voorheen was dit niet zo. Hier hebben we nu van geleerd dat wij dat niet meer kunnen doen, vanwege wij niet mensen willen weg sturen na een week of twee. We zoeken gemotiveerde programmeurs die open staan om nieuwe dingen te leren. Mensen die leergierig en niet bang zijn om een foutje te maken en toe te geven.

**Comment by nathan-vdp** [comment]
Oké top, Ik ga vandaag verder debuggen.

**Comment by nathan-vdp** [comment]
Ik vind het trouwens wel tof hoe jij graag wil helpen, thanks daarvoor!

**Comment by arthurvanl** [comment]
Ik wil graag goeie stagiaires :)

**Comment by nathan-vdp** [comment]
Hey Arthur zou jij misschien kunnen pullen en kijken of de 1ste probleem is gefixt (met een extensie zoeken) als het goed is moet dat nu werken aan mijn kant werkt het.

**Comment by arthurvanl** [comment]
> Hey Arthur zou jij misschien kunnen pullen en kijken of het 1ste probleem is gefixt (met een extensie zoeken) als het goed is moet dat nu werken aan mijn kant werkt het.
Werkt!

**Comment by arthurvanl** [comment]
![image](https://github.com/nathan-vdp/stage-opdracht/assets/20114845/ae07c522-b918-4079-bd42-e7308c211e4d)
Hey maar kijk, ik zie nu dat er iets in mn winkelman zit. Je krijgt alleen niet echt een melding van dat er een domein toegevoegd is aan de winkelmand.

**Comment by arthurvanl** [comment]
![image](https://github.com/nathan-vdp/stage-opdracht/assets/20114845/e88c42ae-e252-436f-875f-690d700ceedb)
Wanneer ik op `Bestellen` druk op de winkelwagen pagina. Wordt er ook een bestelling gemaakt. De winkelmand wordt alleen niet gecleared (krijg je weer extra punten voor)

**Comment by nathan-vdp** [comment]
Oh, top dat alles nu wel werkt. Het is misschien wel handig als ik een alert erin zet wanneer iets in de mand wordt gezet :). Ik had alleen een melding wanneer een domein als actief is.

**Comment by nathan-vdp** [comment]
En over het gesprek: ik wil graag op gesprek komen. Zou dat ook, als dat uitkomt voor jullie natuurlijk, woensdag kunnen? Vrijdag zou eventueel ook kunnen, dan heb ik altijd vrij.

> Ga ik doen. Even over dit:
> 
> > Nou dan heb ik nog wat te doen en, betreft de sollicitatie process ik neem aan dat deze opdracht een vereist is dat je alles hebt gemaakt voor een potentiële sollicitatiegesprek?
> 
> Ik heb zojuist overleg gehad met de eigenaar. We willen graag morgen je gaan beoordelen ookal is de opdracht nog niet af. Dus dat betekent dat je wel gewoon nog op gesprek kan komen. We beoordelen gewoon wat af is en wat niet af is zal niet op gehamert worden.

**Comment by arthurvanl** [comment]
Oh nee, dat gesprek zal veel later pas plaats kunnen komen helaas. Dit vanwege vakanties en andere lopende projecten

**Comment by nathan-vdp** [comment]
Oh jammer, maar dat snap ik. Ik denk ook niet of je al een timeframe kan geven? Ik zelf heb 11 juli vakantie, maar in de tussentijd heb ik alle tijd om een keer langs te komen als dat voor jullie dan ook uitkomt.

**Comment by arthurvanl** [comment]
Nee dat zal ergens in augustus pas zijn. Nog een vraagje trouwens. Zou je de repository op private kunnen zetten. Dit met reden zodat mensen niet gaan opzoeken hoe ze opdracht kunnen maken.
